### PR TITLE
Adding address validation

### LINKED ISSUE #90: Feat: Address Validation / Conversion to Lat/Long

### PR DESCRIPTION
**Pull request by bpalazzi512**
# Description
[Link to Ticket](https://github.com/GenerateNU/skillspark/issues/90)
Adding method for validating addresses through the opencage api. Created endpoints to directly access it, alongside a service that handlers can access internally.

# How Has This Been Tested?
Unit tested + manual route testing

## Screenshots
Screenshot 2026-03-22 at 8 00 43 PM
Screenshot 2026-03-22 at 8 03 36 PM
Screenshot 2026-03-22 at 8 03 06 PM

# Checklist

## General
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [x] I have written unit tests for my code
- [x] I have written integration tests where appropriate
- [x] New and existing unit tests pass locally with my changes
- [x] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [x] I have tested error handling and edge cases
- [x] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [ ] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [ ] Mobile responsive design verified (if applicable)
- [ ] I have run make generate-api to keep the API hooks up to date
- [ ] **For mobile features:** Tested on both iOS and Android devices
- [ ] **For mobile features:** Screenshots from both iOS and Android included below
- [ ] No console errors or warnings
- [ ] Loading states and error states handled

### iOS
[Add iOS screenshots here]

### Android
[Add Android screenshots here]

### CODE REVIEW

File: backend/internal/supabase/migrations/20260317000000_add_geocode_cache.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE geocode_cache (
+    address     TEXT PRIMARY KEY,
```
[the remark this answers is not in the document]
Comment (adescoteaux1) [reply backend/internal/supabase/migrations/20260317000000_add_geocode_cache.sql]: yeah, probably unlikely that we would have much requiring a cache i think

**Review by josh-torre — COMMENTED** [review]
I am not sure I agree with the design here. To be fair we didn't give much direction but I dont think we want to expose the geocoding functionality on our API. Instead, I think we should remove the routes and call the created handler inside the create/update endpoints for locations. If the location exists, we should update the associated lat and long. If the location cannot be determined then we should reject the request to create/update the location and send an error back to the client.

@adescoteaux1 let me know what you think here

All in all great work though :)

**Comment by adescoteaux1** [comment]
> > I am not sure I agree with the design here. To be fair we didn't give much direction but I dont think we want to expose the geocoding functionality on our API. Instead, I think we should remove the routes and call the created handler inside the create/update endpoints for locations. If the location exists, we should update the associated lat and long. If the location cannot be determined then we should reject the request to create/update the location and send an error back to the client.
> > @adescoteaux1 let me know what you think here
> > All in all great work though :)
> 
> it might be good to have the routes so we could validate the address before creation, on blur type thing. That way if they input an invalid address we can tell them before they try sybmitting, without causing an error. Depends on how we want to use it though

Yes good to keep route, but also can we add the validation check to the create and update organization routes

**Review by adescoteaux1 — COMMENTED** [review]
lwk lgtm

**Comment by adescoteaux1** [comment]
> lwk lgtm

oh can we regen frontend api as well

**Review by josh-torre — DISMISSED** [review]
LGTM 🚀

**Review by adescoteaux1 — APPROVED** [review]
